content/docs/getting-started/hello-features.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "Tokio Features"
+weight : 1020
+menu:
+  docs:
+    parent: getting_started
+---
+
+Feel free to skip this section, and just use `features = ["full"]` while
+learning. This section provides some examples of how and when to specify
+tokio features. 
+
+## Often using all features is fine 
+
+When writing an app, we'll often use all of Tokio's features to accelerate
+our development time:
+
+```toml
+[dependencies]
+tokio = { version = "0.2", features = ["full"] }
+```
+
+For most apps, the additional compile time is hardly noticeable relative to the
+cognitive overhead of thinking about which features are needed.
+
+# Selecting features
+
+When we develop protocol implementations that are used by client apps and
+servers, we want to limit the dependencies to decrease compile time --
+not just for us, but also for the users of our crates.
+
+Let's take a look at our tiny client app and see what features are needed.
+
+## tokio main
+
+If we change our code, so that it only uses the `#[tokio::main]` macro:
+
+```
+#[tokio::main]
+async fn main() {
+    println!("doing nothing yet");
+}
+```
+
+then we can choose whether to use threads (commonly used for servers):
+
+```
+tokio = { version = "0.2", features = ["macros", "rt-threaded"] }
+```
+
+or lightweight cooperative multi-tasking
+(often required for low-profile clients): 
+
+```
+tokio = { version = "0.2", features = ["macros", "rt-core"] }
+```
+
+## TcpStream connect
+
+As we start to build the app, we'll need more features.  The line of code
+where we call `TcpStream::connect` requires two features: 
+
+
+```
+    let stream = TcpStream::connect("127.0.0.1:6142").await.unwrap();
```

Comment:
Add `mut` (for writing later).

Reply:
Well for some reason mut is not needed in the hello-world app example in this PR

```
use tokio::net::TcpStream;
use tokio::prelude::*;

#[tokio::main]
async fn main() {
    let mut stream = TcpStream::connect("127.0.0.1:6142").await.unwrap();
    println!("created stream");

    let result = stream.write(b"hello world\n").await;
    println!("wrote to stream; success={:?}", result.is_ok());
}
```

I'll have to look back at other places I needed to declare `mut` in order to write. Hmmm.

Reply:
Not sure what you mean? The hello world example has `mut`. If I take it out, I get:

```
[hello-world (master)]$ cargo run 
   Compiling hello-world v0.1.0 (/Users/jenmei/src/rust/tmp/hello-world)
error[E0596]: cannot borrow `stream` as mutable, as it is not declared as mutable
 --> src/main.rs:9:16
  |
6 |   let stream = TcpStream::connect("127.0.0.1:6142").await.unwrap();
  |       ------ help: consider changing this to be mutable: `mut stream`
```

Reply:
sorry about that, you are correct!